plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/TrinoCatalog.java

```diff
@@ -0,0 +1,101 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.iceberg;
+
+import io.trino.spi.connector.ConnectorMaterializedViewDefinition;
+import io.trino.spi.connector.ConnectorSession;
+import io.trino.spi.connector.ConnectorViewDefinition;
+import io.trino.spi.connector.SchemaTableName;
+import io.trino.spi.connector.TableNotFoundException;
+import io.trino.spi.security.TrinoPrincipal;
+import org.apache.iceberg.PartitionSpec;
+import org.apache.iceberg.Schema;
+import org.apache.iceberg.Table;
+import org.apache.iceberg.Transaction;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+/**
+ * An interface to allow different Iceberg catalog implementations in IcebergMetadata.
+ * <p>
+ * It mimics the Iceberg catalog interface, with the following modifications:
+ * <ul>
+ *   <li>ConnectorSession is added at the front of each method signature</li>
+ *   <li>String is used to identify namespace instead of Iceberg Namespace, Optional.empty() is used to represent Namespace.empty().
+ *      This delegates the handling of multi-level namespace to each implementation</li>
+ *   <li>Similarly, SchemaTableName is used to identify table instead of Iceberg TableIdentifier</li>
+ *   <li>Metadata is a map of string to object instead of string to string</li>
+ *   <li>Additional methods related to authorization are added</li>
+ *   <li>View related methods are currently mostly the same as ones in ConnectorMetadata.
+ *      These methods will likely be updated once Iceberg view interface is added.</li>
+ * </ul>
+ */
+public interface TrinoCatalog
+{
+    List<String> listNamespaces(ConnectorSession session);
+
+    boolean dropNamespace(ConnectorSession session, String namespace);
+
+    Map<String, Object> loadNamespaceMetadata(ConnectorSession session, String namespace);
+
+    Optional<TrinoPrincipal> getNamespacePrincipal(ConnectorSession session, String namespace);
+
+    void createNamespace(ConnectorSession session, String namespace, Map<String, Object> map, TrinoPrincipal owner);
+
+    void setNamespacePrincipal(ConnectorSession session, String namespace, TrinoPrincipal principal);
+
+    void renameNamespace(ConnectorSession session, String source, String target);
+
+    List<SchemaTableName> listTables(ConnectorSession session, Optional<String> namespace);
+
+    Transaction newCreateTableTransaction(ConnectorSession session, SchemaTableName schemaTableName, Schema schema, PartitionSpec partitionSpec, String location,
+            Map<String, String> properties);
+
+    boolean dropTable(ConnectorSession session, SchemaTableName schemaTableName, boolean purge);
+
+    void renameTable(ConnectorSession session, SchemaTableName from, SchemaTableName to);
+
+    Table loadTable(ConnectorSession session, SchemaTableName schemaTableName) throws TableNotFoundException, UnknownTableTypeException;
```

Comment:
I'm not sure how useful the exception declaration here is. Since they're RTE, they aren't enforced on the implementing class signatures or accessors. I think it'd be better to mention them as a javadoc.

Reply:
sure, will do